js/src/table.ts

```diff
@@ -16,280 +16,226 @@
-    }
+    /**
+     * @summary Set an element value by position.
+     * @param index The index of the element to write.
+     * @param value The value to set.
+     */
+    // @ts-ignore
+    public set(index: number, value: T['TValue'] | null): void { return; }
 
-    protected _schema: Schema<T>;
-    // List of inner RecordBatches
-    protected _chunks: RecordBatch<T>[];
-    protected _children?: Column<T[keyof T]>[];
+    /**
+     * @summary Retrieve the index of the first occurrence of a value in an Vector.
+     * @param element The value to locate in the Vector.
+     * @param offset The index at which to begin the search. If offset is omitted, the search starts at index 0.
+     */
+    // @ts-ignore
+    public indexOf(element: T['TValue'], offset?: number): number { return -1; }
 
-    public get schema() { return this._schema; }
-    public get length() { return this._length; }
-    public get chunks() { return this._chunks; }
-    public get numCols() { return this._numChildren; }
+    /**
+     * @summary Get the size in bytes of an element by index.
+     * @param index The index at which to get the byteLength.
+     */
+    // @ts-ignore
+    public getByteLength(index: number): number { return 0; }
 
-    public clone(chunks = this._chunks) {
-        return new Table<T>(this._schema, chunks);
+    /**
+     * @summary Iterator for rows in this Table.
+     */
+    public [Symbol.iterator]() {
+        return new ChunkedIterator(this.data);
     }
 
-    public getColumn<R extends keyof T>(name: R): Column<T[R]> {
-        return this.getColumnAt(this.getColumnIndex(name)) as Column<T[R]>;
-    }
-    public getColumnAt<R extends DataType = any>(index: number): Column<R> | null {
-        return this.getChildAt(index);
-    }
-    public getColumnIndex<R extends keyof T>(name: R) {
-        return this._schema.fields.findIndex((f) => f.name === name);
-    }
-    public getChildAt<R extends DataType = any>(index: number): Column<R> | null {
-        if (index < 0 || index >= this.numChildren) { return null; }
-        let field: Field<R>, child: Column<R>;
-        const fields = (this._schema as Schema<any>).fields;
-        const columns = this._children || (this._children = []) as Column[];
-        if (child = columns[index]) { return child as Column<R>; }
-        if (field = fields[index]) {
-            const chunks = this._chunks
-                .map((chunk) => chunk.getChildAt<R>(index))
-                .filter((vec): vec is Vector<R> => vec != null);
-            if (chunks.length > 0) {
-                return (columns[index] = new Column<R>(field, chunks));
-            }
-        }
-        return null;
+    /**
+     * @summary Return a JavaScript Array of the Table rows.
+     * @returns An Array of Table rows.
+     */
+    public toArray() {
+        return this.data.reduce((ary, data) =>
+            ary.concat(toArrayVisitor.visit(data)),
+            new Array<Struct<T>['TValue']>()
+        );
```

Comment:
How does this perform compared to `return [...this]`?

Reply:
dunno, can't run benchmarks yet :stuck_out_tongue: